### ISSUE TITLE
Refactor: Append interpolated string to Append calls

### ISSUE THREAD

**Issue by wmjordan** [dotnet/roslynator#1562]
Given the following code:

````csharp
var sb = new StringBuilder();
var var1 = DateTime.Now.ToString();

sb.Append($"ABC{var1}EFG");
````

The third line used string interpolation, which is unnecessary, can be refactored to a series of `Append` calls as the following code shows.

````csharp
sb.Append("ABC").Append(var1).Append("EFG");
````

**Comment by josefpihrt** [comment]
Following code will use `AppendInterpolatedStringHandler` under the hood which is even better then `StringBuilder.Append` methods.
```
sb.Append($"ABC{var1}EFG");
```

SharpLab: https://sharplab.io/#v2:CYLg1APgAgTAjAWAFBQAwAIpwHQBUCmAHgC4Dcyys6AwsgN7LpOYAs6AsgBQCUjzDSZkPQA3AIYAndAGcARugC86AHb4A7ugDKxCQEtlAcwBCAV10AbYPgk9yg4U3FSncRegAiY4vly6AtvjYAHIA9mp4Idp6hrYU9g5y2ACCAA4p+MrAnAAkAERJRtR0LgC+AKIAYgDiudx2QiXIjUhAA==

Following code will be converted into previous case by [RCS1267](https://josefpihrt.github.io/docs/roslynator/analyzers/RCS1267).
```
sb.Append(String.Concat("ABC", var1, "EFG"));
```

Following code could be added to [RCS1267](https://josefpihrt.github.io/docs/roslynator/analyzers/RCS1267) as an improvement.
```
sb.Append("ABC" + var1 + "EFG");
```

**Comment by wmjordan** [comment]
Yes, I know that the AppendInterpolatedStringHandler thing.
However, in my practical situation, there are quite a few `Append` calls around this one. Thus the `StringBuilder` can not be simply substituted by a interpolated string. For example,

````csharp
var sb = new StringBuilder();
var var1 = DateTime.Now.ToString();

for (var i = 0; i = 100; i++) {
    sb.Append(somethingElse(i));
}
sb.Append($"ABC{var1}EFG");

````

or sometimes, the `sb.Append(interpolated string)` is invoked within a loop. So, it is arguable that the string interpolation will bring better benefit to the overall performance.

Leaving the AppendInterpolatedStringHandler thing there, will generate a lot of intermediate string instances, cause extra memory copying, and put burden to the GC. If we break it into `Append` calls, there will be no such burden then.

**Comment by josefpihrt** [comment]
> Thus the StringBuilder can not be simply substituted by a interpolated string.

Not sure if I understand. Why it couldn't be substitued?

> sb.Append(interpolated string) is invoked within a loop

Why it matters if it's in a loop or not?

> will generate a lot of intermediate string instances, cause extra memory copying

Where do you see these allocations in the SharpLab example?

**Comment by wmjordan** [comment]
You are right. From .NET 6 on, there is no significant performance degrade when calling StringBuilder.Append(interpolated string).

Sorry, I did not mentioned that the situation was that the code was compiled for the old .NET Framework where no AppendInterpolatedStringHandler exists. I checked the decompiled code from an assembly and the code is compiled as a call to String.Format, then StringBuilder.Append(string).
